x-pack/plugins/cases/public/components/edit_connector/index.tsx

```diff
@@ -290,36 +242,47 @@ export const EditConnector = React.memo(
             <EuiFlexItem grow={false} data-test-subj="connector-edit-header">
               <h4>{i18n.CONNECTORS}</h4>
             </EuiFlexItem>
-            {isLoading && <EuiLoadingSpinner data-test-subj="connector-loading" />}
-            {!isLoading && !editConnector && permissions.push && actionsReadCapabilities && (
-              <EuiFlexItem data-test-subj="connector-edit" grow={false}>
-                <EuiButtonIcon
-                  data-test-subj="connector-edit-button"
-                  aria-label={i18n.EDIT_CONNECTOR_ARIA}
-                  iconType={'pencil'}
-                  onClick={onEditClick}
+            <EuiFlexItem data-test-subj="connector-edit" grow={false}>
+              <EuiButtonIcon
+                data-test-subj="connector-edit-button"
+                aria-label={i18n.EDIT_CONNECTOR_ARIA}
+                iconType="pencil"
+                onClick={onEditClick}
+                disabled={
+                  isLoading || editConnector || !hasPushPermissions || !actionsReadCapabilities
+                }
+              />
+            </EuiFlexItem>
+          </EuiFlexGroup>
+          <EuiHorizontalRule margin="xs" />
+          <EuiFlexGroup
+            data-test-subj="edit-allAvailableConnectors"
+            direction="column"
+            alignItems="stretch"
+          >
+            {!isLoading && !editConnector && hasErrorMessages && actionsReadCapabilities && (
+              <EuiFlexItem data-test-subj="push-callouts">
+                <PushCallouts
+                  errorsMsg={errorsMsg}
+                  hasLicenseError={hasLicenseError}
+                  hasConnectors={allAvailableConnectors.length > 0}
+                  onEditClick={onEditClick}
                 />
               </EuiFlexItem>
             )}
-          </MyFlexGroup>
-          <EuiHorizontalRule margin="xs" />
-          <MyFlexGroup data-test-subj="edit-connectors" direction="column">
-            {!isLoading && !editConnector && pushCallouts && actionsReadCapabilities && (
-              <EuiFlexItem data-test-subj="push-callouts">{pushCallouts}</EuiFlexItem>
-            )}
-            <DisappearingFlexItem $isHidden={!editConnector}>
```

Comment:
No need to hide the component.

Reply:
Currently if user has only `read permission / viewer role`, and there are `no connectors added to case` then user sees only header `External Incident Management System`.
![Screenshot 2023-01-26 at 11 35 26](https://user-images.githubusercontent.com/117571355/214816633-4d694079-04e4-465c-90c6-5be08609d168.png)

Is it expected behaviour or it should be hidden?